engine/lite/kafka/api/src/main/scala/pl/touk/nussknacker/engine/lite/kafka/api/LiteKafkaSource.scala

```diff
@@ -1,31 +1,36 @@
 package pl.touk.nussknacker.engine.lite.kafka.api
 
 import cats.Monad
-import cats.data.Validated.Valid
-import cats.data.ValidatedNel
+import cats.data.{Validated, ValidatedNel}
 import org.apache.kafka.clients.consumer.ConsumerRecord
+import pl.touk.nussknacker.engine.api.exception.NuExceptionInfo
 import pl.touk.nussknacker.engine.api.runtimecontext.EngineRuntimeContext
-import pl.touk.nussknacker.engine.api.{Context, Lifecycle}
+import pl.touk.nussknacker.engine.api.{Context, ContextId, Lifecycle}
 import pl.touk.nussknacker.engine.lite.api.commonTypes.ErrorType
 import pl.touk.nussknacker.engine.lite.api.customComponentTypes
 import pl.touk.nussknacker.engine.lite.api.customComponentTypes.LiteSource
 
 import scala.language.higherKinds
+import scala.util.Try
 
 trait LiteKafkaSource extends LiteSource[ConsumerRecord[Array[Byte], Array[Byte]]] with Lifecycle {
 
   def topics: List[String]
 
-  private var context: EngineRuntimeContext = _
+  protected var context: EngineRuntimeContext = _
 
   override def open(context: EngineRuntimeContext): Unit = {
     this.context = context
   }
 
-  override def createTransformation[F[_] : Monad](evaluateLazyParameter: customComponentTypes.CustomComponentContext[F]): ConsumerRecord[Array[Byte], Array[Byte]] => ValidatedNel[ErrorType, Context] = record =>
-    Valid(deserialize(context, record))
+  protected def nextContextId: ContextId
```

Comment:
I don't think I like this exposing nextContextId, but I don't have a better idea now, so let's leave it...

Reply:
I also don't like it. The problem was that `ContextInitializingFunction` was coupled with `ContextIdGenerator` and it was done because of Flink serialization. I've decoupled them and it looks better now. Can you take a look?